pydantic/types.py

```diff
@@ -838,3 +842,97 @@ def to(self, unit: str) -> float:
             raise errors.InvalidByteSizeUnit(unit=unit)
 
         return self / unit_div
+
+
+if not TYPE_CHECKING:  # noqa: C901
+    # Actual implementation, Cythonable
+    class SplitStr(list):
+        # Needed for pydantic to detect that this is a list
+        __origin__ = list
+        __args__: List[Type[T]]
+        item_type: Type[T]
+
+        # classmethod not required, but make Cython happy until 0.30.0 is available
+        # ref: https://github.com/cython/cython/issues/3764
+        @classmethod
+        def __class_getitem__(cls, item_type: Type[T]) -> Type[List[T]]:
+            # __args__ is needed to conform to typing generics api
+            namespace = {'item_type': item_type, '__args__': [item_type]}
+            # We use new_class to be able to deal with Generic types
+            return new_class(f'{cls.__name__}Type', (cls,), {}, lambda ns: ns.update(namespace))
+
+        @classmethod
+        def __get_validators__(cls) -> 'CallableGenerator':
+            yield cls.split_str_validator
+
+        @classmethod
+        def __modify_schema__(cls, field_schema: Dict[str, Any]) -> None:
+            str_schema = {**field_schema, 'type': 'string'}
+            if 'items' in str_schema:
+                del str_schema['items']
+            any_of = [field_schema.copy(), str_schema]
+            field_schema.clear()
+            field_schema['anyOf'] = any_of
+
+        @classmethod
+        def split_str_validator(cls, v: 'Optional[Union[List[T], str]]', field: 'ModelField') -> 'Optional[List[T]]':
+            if v is None and not field.required:
+                return None
+            if isinstance(v, str):
+                use_v: List[Any] = v.split()
```

Comment:
This will split on newlines as well, which might not be desirable (though I do want it because dotenv allows multiline strings). Without an explicit argument to `split()`, it'll drop empty values, which won't match the behaviour of commas (either for `split` or `csv` parsing).

If we switch to using `csv`, it doesn't look like you can get both newlines and spaces at the same time (would you even want to?) - this means potentially introducing new types (beyond what exists here) for newline delimiters.